theme/static/js/hs_file_browser.js

```diff
@@ -1243,6 +1244,69 @@ function updateNavigationState() {
     $("#fb-move-up").toggleClass("disabled", !(getCurrentPath().path.length || getCurrentPath().hasOwnProperty("aggregation")));    // The root path is an empty string
 }
 
+function anyExternalFiles(fullPaths) {
+    /*
+    Look for any .url files.
+
+    param: fullPaths array of file path strings to parse top level folders and file extensions from
+     */
+    let filePathsContainingUrl = [];
+
+    for (const [k, v] of Object.entries(fullPaths)) {
+        if (v.file_name.split('.').reverse()[0].toLowerCase() === ('url')) {
+            filePathsContainingUrl.push(v.url)
+        }
+    }
+
+    return filePathsContainingUrl.length > 0
+}
+
+function countSelectedExternalFiles(fullPaths) {
+    /*
+    Based on full file paths enumerate .url files from subfolders and implied current path from Content manager.
+    Comparison from selections is based on the selections in the Content manager at the time this function is invoked.
+    Parent folders are not counted.
+
+    param: fullPaths array of file path strings to parse top level folders and file extensions from
+     */
+    let filePathsContainingUrl = [];
+    let topFolders = {};
+    let selectedFiles = [];
+    let numUrlWithinSubs = 0;
+
+    for (const [k, v] of Object.entries(fullPaths)) {
+        if (v.file_name.split('.').reverse()[0].toLowerCase() === ('url')) {
+            filePathsContainingUrl.push(v.url)
+        }
+    }
+
+    // enumerate external links in subfolders
+    filePathsContainingUrl.forEach(function(path) {
+        let ele_full = path.split('contents').reverse()[0] // for example /testfolder/extcontent.url
+        let ele = ele_full.split('/')[1] // for example testfolder
+        if (topFolders[ele]) {
+            topFolders[ele] = topFolders[ele] + 1
+        } else {
+            topFolders[ele] = 1
+        }
+    })
+
+    $("#fb-files-container li.ui-selected").each(function(i, el) {
+        if (this.title.includes('Type: File Folder')) { // review subfolder info
+            let ele = String(this.innerText).trim();
+            if (topFolders[ele]) {
+                numUrlWithinSubs += topFolders[ele];
+            }
+        }
+
+        // look at current directory
+        if (this.innerText.toLowerCase().includes('.url')) {
+            selectedFiles.push(this.innerText);
+        }
+    });
+    return selectedFiles.length + numUrlWithinSubs;
+}
+
```

Comment:
This can be handled on the server side rather than checking external url files on the client side. This looks like to be related to the email thread on referenced file aggregation and should be easily achievable on the server side.

Reply:
New endpoint and logic